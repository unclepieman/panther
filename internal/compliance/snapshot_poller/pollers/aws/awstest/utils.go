package awstest

/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import (
	"time"

	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/arn"
)

// Example output from the AWS API
var (
	ExampleTime, _             = time.Parse(time.RFC3339, "2019-04-02T17:16:30+00:00")
	ExampleIntegrationID       = aws.String("8e39aa9d-9823-4872-a1bd-40fd8795634b")
	ExampleAuthSource          = "arn:aws:iam::123456789012:role/PantherAuditRole"
	ExampleAuthSourceParsedARN = ParseExampleAuthSourceARN(ExampleAuthSource)
	ExampleAccountId           = aws.String("123456789012")
	ExampleRegion              = aws.String("us-west-2")
)

// ParseExampleAuthSourceARN returns a parsed Auth Source ARN
func ParseExampleAuthSourceARN(arnToParse string) arn.ARN {
	parsedArn, err := arn.Parse(arnToParse)
	if err != nil {
		return arn.ARN{}
	}

	return parsedArn
}
