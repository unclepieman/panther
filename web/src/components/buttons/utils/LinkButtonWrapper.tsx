/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
import React from 'react';
import { Box, LinkProps } from 'pouncejs';
import { Link as RRLink } from 'react-router-dom';

export type LinkButtonWrapperProps = Pick<LinkProps, 'external' | 'disabled' | 'to'>;

const LinkButton: React.FC<LinkButtonWrapperProps> = ({ disabled, external, to, children }) => {
  let linkProps: LinkProps;
  if (disabled) {
    linkProps = { as: 'span' as React.ElementType };
  } else if (!external) {
    linkProps = { to, as: RRLink };
  } else {
    linkProps = {
      target: '_blank',
      rel: 'noopener noreferrer',
      href: to as string,
      as: 'a' as React.ElementType,
    };
  }
  return (
    <Box
      {...linkProps}
      sx={{
        '& > span': {
          display: 'flex',
          justifyContent: 'center',
          alignItems: 'center',
        },
      }}
    >
      {children}
    </Box>
  );
};
export default LinkButton;
