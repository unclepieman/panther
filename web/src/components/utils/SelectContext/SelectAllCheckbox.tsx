/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import { useSelect } from 'Components/utils/SelectContext/SelectContext';
import { Checkbox } from 'pouncejs';

interface SelectAllCheckboxProps<T> {
  selectionItems: T[];
}

function SelectAllCheckboxComponent<T>({ selectionItems, ...rest }: SelectAllCheckboxProps<T>) {
  const { selection, resetSelection, selectAll } = useSelect<T>();
  return (
    <Checkbox
      checked={!!selection.length}
      onChange={() => {
        if (selection.length) {
          resetSelection();
        } else {
          selectAll(selectionItems);
        }
      }}
      aria-label={selection.length ? 'unselect all' : 'select all'}
      {...rest}
    />
  );
}

export default React.memo(SelectAllCheckboxComponent);
