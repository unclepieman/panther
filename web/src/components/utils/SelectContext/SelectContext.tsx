/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React, { ReactNode } from 'react';

const identity = (x: any) => x;

export interface SelectContextValue<T> {
  selection: T[];
  selectItem: (item: T) => void;
  deselectItem: (item: T) => void;
  resetSelection: () => void;
  selectAll: (items: T[]) => void;
  checkIfSelected: (item: T) => boolean;
  toggleItem: (item: T) => void;
}

const SelectContext = React.createContext<SelectContextValue<any>>(undefined);

interface SelectProviderProps<T> {
  children: React.ReactNode;
  initialSelection?: T[];
  getItemKey?: (item: T) => string;
}

function SelectProvider<T>({
  initialSelection = [],
  getItemKey = identity,
  children,
}: SelectProviderProps<T>) {
  const [selection, setSelected] = React.useState<Array<T>>(initialSelection);

  /**
   * @public
   * Add an item to the selection
   *
   */
  const selectItem = React.useCallback(item => {
    return setSelected(existing => [...existing, item]);
  }, []);

  /**
   * @public
   * Deselects an item from the selection
   *
   */
  const deselectItem = React.useCallback(
    item => {
      return setSelected(existing => existing.filter(i => getItemKey(i) !== getItemKey(item)));
    },
    [getItemKey]
  );

  /**
   * @public
   * Reset selection to an empty array
   *
   */
  const resetSelection = React.useCallback(() => setSelected([]), []);

  const selectAll = React.useCallback((items: T[]) => {
    return setSelected(items);
  }, []);

  /**
   * @public
   * Simple function that checks whether an item is selected
   *
   */
  const checkIfSelected = React.useCallback(
    (item: T) => {
      return !!selection.find(i => getItemKey(i) === getItemKey(item));
    },
    [selection, getItemKey]
  );

  /**
   * @public
   * This function check whether an item is selected
   * and change its status to the opposite
   */
  const toggleItem = React.useCallback(
    (item: T) => {
      return checkIfSelected(item) ? deselectItem(item) : selectItem(item);
    },
    [checkIfSelected, deselectItem, selectItem]
  );

  const contextValue = React.useMemo(
    () => ({
      selection,
      selectAll,
      deselectItem,
      selectItem,
      resetSelection,
      checkIfSelected,
      toggleItem,
    }),
    [selection, resetSelection, selectAll, selectItem, deselectItem, checkIfSelected, toggleItem]
  );

  return <SelectContext.Provider value={contextValue}>{children}</SelectContext.Provider>;
}

const MemoizedSelectProvider = React.memo(SelectProvider);

const withSelectContext = <T,>(config?: Omit<SelectProviderProps<T>, 'children'>) => (
  Component: React.FC
) => props => (
  <SelectProvider {...config}>
    <Component {...props} />
  </SelectProvider>
);

const useSelect = <T extends unknown = string>() =>
  React.useContext<SelectContextValue<T>>(SelectContext);

/** A shortcut for the consumer component */

const SelectConsumer = <T extends unknown = string>(
  props: React.ConsumerProps<SelectContextValue<T>>
) => {
  const Consumer = SelectContext.Consumer as React.Consumer<SelectContextValue<T>>;
  return <Consumer {...props} />;
};

export {
  SelectContext,
  SelectConsumer,
  MemoizedSelectProvider as SelectProvider,
  withSelectContext,
  useSelect,
};
