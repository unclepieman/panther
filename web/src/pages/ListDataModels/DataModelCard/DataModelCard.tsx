/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import { Flex, Link, Badge, Box } from 'pouncejs';
import { DataModel } from 'Generated/schema';
import GenericItemCard from 'Components/GenericItemCard';
import { Link as RRLink } from 'react-router-dom';
import urls from 'Source/urls';
import { formatDatetime } from 'Helpers/utils';
import BulletedValue from 'Components/BulletedValue';
import { SelectCheckbox } from 'Components/utils/SelectContext';
import DataModelCardOptions from './DataModelCardOptions';

interface DataModelCardProps {
  dataModel: DataModel;
  selectionEnabled?: boolean;
  isSelected?: boolean;
}

const DataModelCard: React.FC<DataModelCardProps> = ({
  dataModel,
  isSelected = false,
  selectionEnabled = false,
}) => {
  return (
    <GenericItemCard isHighlighted={isSelected}>
      {selectionEnabled && (
        <Box transform="translate3d(-12px,-12px,0)">
          <SelectCheckbox selectionItem={dataModel} />
        </Box>
      )}
      <GenericItemCard.Body>
        <GenericItemCard.Header>
          <GenericItemCard.Heading>
            <Link as={RRLink} to={urls.logAnalysis.dataModels.details(dataModel.id)}>
              {dataModel.displayName || dataModel.id}
            </Link>
          </GenericItemCard.Heading>
          <GenericItemCard.HeadingValue label="ID" value={dataModel.id} labelFirst withDivider />
          <GenericItemCard.HeadingValue
            value={formatDatetime(dataModel.lastModified)}
            label="Updated"
            labelFirst
          />
          <DataModelCardOptions dataModel={dataModel} />
        </GenericItemCard.Header>

        <GenericItemCard.ValuesGroup>
          <GenericItemCard.Value
            label="Log Type"
            value={<BulletedValue value={dataModel.logTypes[0]} />}
          />
          <Flex ml="auto" mr={0} align="flex-end" spacing={4}>
            <Badge color={dataModel.enabled ? 'cyan-400' : 'navyblue-300'}>
              {dataModel.enabled ? 'ENABLED' : 'DISABLED'}
            </Badge>
          </Flex>
        </GenericItemCard.ValuesGroup>
      </GenericItemCard.Body>
    </GenericItemCard>
  );
};

export default React.memo(DataModelCard);
