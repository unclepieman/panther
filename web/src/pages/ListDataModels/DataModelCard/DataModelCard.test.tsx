/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import { SelectProvider } from 'Components/utils/SelectContext';
import { buildDataModel, render } from 'test-utils';
import DataModelCard from './DataModelCard';

describe('DataModelCard', () => {
  it('matches snapshot', () => {
    const { container } = render(<DataModelCard dataModel={buildDataModel()} />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with selection', () => {
    const { container } = render(
      <SelectProvider>
        <DataModelCard dataModel={buildDataModel()} selectionEnabled />
      </SelectProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders the necessary information', () => {
    const dataModel = buildDataModel();

    const { getByText, getByAriaLabel } = render(
      <SelectProvider>
        <DataModelCard dataModel={dataModel} />
      </SelectProvider>
    );

    expect(getByText(dataModel.id)).toBeInTheDocument();
    expect(getByText(dataModel.displayName)).toBeInTheDocument();
    expect(getByText(dataModel.enabled ? 'ENABLED' : 'DISABLED')).toBeInTheDocument();
    expect(getByAriaLabel('Toggle Options')).toBeInTheDocument();
  });

  it('fallbacks to `id` when display name is not existent', () => {
    const dataModel = buildDataModel({ displayName: '' });

    const { getAllByText } = render(
      <SelectProvider>
        <DataModelCard dataModel={dataModel} />
      </SelectProvider>
    );

    expect(getAllByText(dataModel.id)).toHaveLength(2);
  });
});
