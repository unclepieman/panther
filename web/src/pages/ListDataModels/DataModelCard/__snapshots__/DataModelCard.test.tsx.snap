// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataModelCard matches snapshot 1`] = `
.panther-23 {
  background-color: #1d2a3f;
  border-radius: 4px;
  overflow: hidden;
}

.panther-22 {
  background-color: transparent;
  padding: 16px;
}

.panther-21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  height: 100%;
}

.panther-20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
}

.panther-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin-bottom: 8px;
}

.panther-1 {
  font-weight: 500;
  margin-right: auto;
  max-width: 60%;
  word-break: break-word;
}

.panther-0 {
  -webkit-transition: color 0.1s ease-out;
  transition: color 0.1s ease-out;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #7fd3ff;
}

.panther-0:hover {
  color: #ace2ff;
}

.panther-0:focus {
  color: #ace2ff;
}

.panther-0:active,
.panther-0[data-active=true] {
  color: #11a5f3;
}

.panther-3 {
  font-size: 0.75rem;
  max-width: 20%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.panther-2 {
  margin-right: 4px;
  color: #86a3c3;
}

.panther-4 {
  padding-left: 8px;
  padding-right: 8px;
  font-size: 0.75rem;
  color: #86a3c3;
}

.panther-9 {
  margin-left: 8px;
  margin-top: -4px;
}

.panther-8 {
  outline: none;
  line-height: 1;
  border-radius: 99999px;
  height: 24px;
  width: 24px;
  -webkit-transition: background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border: 1px solid;
  border-color: transparent;
  background-color: transparent;
  cursor: pointer;
  color: #f6f6f6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.panther-8:hover {
  background-color: rgba(62,81,107,0.3);
}

.panther-8:focus {
  background-color: rgba(62,81,107,0.3);
}

.panther-8:active,
.panther-8[data-active=true] {
  background-color: #3e516b;
}

.panther-7 {
  display: inline-block;
  vertical-align: sub;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 16px;
  height: 16px;
  color: currentColor;
}

.panther-19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.panther-19>*:not(:last-child) {
  margin-right: 32px;
}

.panther-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 8px;
}

.panther-11 {
  color: #bdbdbd;
  font-size: 0.563rem;
  font-weight: 500;
}

.panther-15 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-size: 0.875rem;
  font-weight: 500;
  opacity: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 24px;
}

.panther-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.panther-14>*:not(:last-child) {
  margin-right: 8px;
}

.panther-12 {
  width: 12px;
  height: 12px;
  background-color: hsl(49, 39%, 69%);
  border-radius: 99999px;
}

.panther-13 {
  font-size: 0.75rem;
  font-weight: 400;
  line-height: 1.5;
}

.panther-18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  margin-left: auto;
  margin-right: 0;
}

.panther-18>*:not(:last-child) {
  margin-right: 16px;
}

.panther-17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: default;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-width: 85px;
  text-align: center;
  font-weight: 700;
  border-radius: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-size: 0.75rem;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  background-color: #3e516b;
}

<div>
  <header
    id="main-header"
  />
  <section
    class="panther-23"
  >
    <div
      class="panther-22"
    >
      <div
        class="panther-21"
      >
        <div
          class="panther-20"
        >
          <header
            class="panther-10"
          >
            <h4
              class="panther-1"
            >
              <a
                class="panther-0"
                href="/log-analysis/data-models/97c37f76-8bd8-4def-b4ab-7bfe83d62081/"
              >
                collaboration
              </a>
            </h4>
            <div
              class="panther-3"
            >
              <span
                class="panther-2"
              >
                ID
              </span>
              97c37f76-8bd8-4def-b4ab-7bfe83d62081
            </div>
            <span
              class="panther-4"
            >
              â€¢
            </span>
            <div
              class="panther-3"
            >
              <span
                class="panther-2"
              >
                Updated
              </span>
              2020-08-20 04:58 GMT
            </div>
            <div
              class="panther-9"
            >
              <button
                aria-controls="menu--1"
                aria-haspopup="true"
                aria-label="Toggle Options"
                aria-pressed="false"
                class="panther-8"
                data-reach-menu-button=""
                id="menu-button--menu--1"
                type="button"
              >
                <svg
                  class="panther-7"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 10C4.9 10 4 10.9 4 12C4 13.1 4.9 14 6 14C7.1 14 8 13.1 8 12C8 10.9 7.1 10 6 10ZM18 10C16.9 10 16 10.9 16 12C16 13.1 16.9 14 18 14C19.1 14 20 13.1 20 12C20 10.9 19.1 10 18 10ZM12 10C10.9 10 10 10.9 10 12C10 13.1 10.9 14 12 14C13.1 14 14 13.1 14 12C14 10.9 13.1 10 12 10Z"
                  />
                </svg>
              </button>
            </div>
          </header>
          <div
            class="panther-19"
          >
            <dl
              class="panther-16"
            >
              <dt
                aria-labelledby="log-typeobject-object"
                class="panther-11"
              >
                Log Type
              </dt>
              <dd
                aria-labelledby="log-typeobject-object"
                class="panther-15"
              >
                <div
                  class="panther-14"
                >
                  <span
                    class="panther-12"
                  />
                  <span
                    class="panther-13"
                  >
                    enterprise
                  </span>
                </div>
              </dd>
            </dl>
            <div
              class="panther-18"
            >
              <div
                aria-atomic="true"
                class="panther-17"
                role="status"
              >
                DISABLED
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <footer
    id="footer"
  />
</div>
`;

exports[`DataModelCard matches snapshot with selection 1`] = `
.panther-27 {
  background-color: #1d2a3f;
  border-radius: 4px;
  overflow: hidden;
}

.panther-26 {
  background-color: transparent;
  padding: 16px;
}

.panther-25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  height: 100%;
}

.panther-24 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
}

.panther-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin-bottom: 8px;
}

.panther-5 {
  font-weight: 500;
  margin-right: auto;
  max-width: 60%;
  word-break: break-word;
}

.panther-4 {
  -webkit-transition: color 0.1s ease-out;
  transition: color 0.1s ease-out;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #7fd3ff;
}

.panther-4:hover {
  color: #ace2ff;
}

.panther-4:focus {
  color: #ace2ff;
}

.panther-4:active,
.panther-4[data-active=true] {
  color: #11a5f3;
}

.panther-7 {
  font-size: 0.75rem;
  max-width: 20%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.panther-6 {
  margin-right: 4px;
  color: #86a3c3;
}

.panther-8 {
  padding-left: 8px;
  padding-right: 8px;
  font-size: 0.75rem;
  color: #86a3c3;
}

.panther-13 {
  margin-left: 8px;
  margin-top: -4px;
}

.panther-12 {
  outline: none;
  line-height: 1;
  border-radius: 99999px;
  height: 24px;
  width: 24px;
  -webkit-transition: background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: background-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border: 1px solid;
  border-color: transparent;
  background-color: transparent;
  cursor: pointer;
  color: #f6f6f6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.panther-12:hover {
  background-color: rgba(62,81,107,0.3);
}

.panther-12:focus {
  background-color: rgba(62,81,107,0.3);
}

.panther-12:active,
.panther-12[data-active=true] {
  background-color: #3e516b;
}

.panther-11 {
  display: inline-block;
  vertical-align: sub;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 16px;
  height: 16px;
  color: currentColor;
}

.panther-23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.panther-23>*:not(:last-child) {
  margin-right: 32px;
}

.panther-20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 8px;
}

.panther-15 {
  color: #bdbdbd;
  font-size: 0.563rem;
  font-weight: 500;
}

.panther-19 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-size: 0.875rem;
  font-weight: 500;
  opacity: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 24px;
}

.panther-18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.panther-18>*:not(:last-child) {
  margin-right: 8px;
}

.panther-16 {
  width: 12px;
  height: 12px;
  background-color: hsl(49, 39%, 69%);
  border-radius: 99999px;
}

.panther-17 {
  font-size: 0.75rem;
  font-weight: 400;
  line-height: 1.5;
}

.panther-22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  margin-left: auto;
  margin-right: 0;
}

.panther-22>*:not(:last-child) {
  margin-right: 16px;
}

.panther-21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: default;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-width: 85px;
  text-align: center;
  font-weight: 700;
  border-radius: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  font-size: 0.75rem;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  background-color: #3e516b;
}

.panther-3 {
  -webkit-transform: translate3d(-12px,-12px,0);
  -moz-transform: translate3d(-12px,-12px,0);
  -ms-transform: translate3d(-12px,-12px,0);
  transform: translate3d(-12px,-12px,0);
}

.panther-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  font-size: 0.875rem;
  font-weight: 500;
  vertical-align: top;
}

.panther-1 {
  position: relative;
  border-radius: 99999px;
  padding: 8px;
  -webkit-transition: background-color 0.15s linear;
  transition: background-color 0.15s linear;
}

.panther-1:hover {
  background-color: rgba(62,81,107,0.2);
}

.panther-1:hover:after {
  border-color: #0081c5;
}

.panther-1:focus-within {
  background-color: rgba(62,81,107,0.2);
}

.panther-1:focus-within:after {
  border-color: #0081c5;
}

.panther-1::before {
  content: url( 'data:image/svg+xml; utf8,<svg xmlns="http://www.w3.org/2000/svg" width="18" height="18" viewBox="0 0 20 18" fill="white"><path d="M7 14.17L2.83 10l-1.41 1.41L7 17 19 5l-1.41-1.42L7 14.17z" /></svg>' );
  display: block;
  position: absolute;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  opacity: 0;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
}

.panther-1::after {
  content: "";
  display: block;
  width: 26px;
  height: 26px;
  border: 1px solid;
  border-radius: 2px;
  -webkit-transition: border-color 125ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 125ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border-color: #3e516b;
}

.panther-0 {
  cursor: pointer;
  position: absolute;
  opacity: 0;
}

<div>
  <header
    id="main-header"
  />
  <section
    class="panther-27"
  >
    <div
      class="panther-26"
    >
      <div
        class="panther-25"
      >
        <div
          class="panther-3"
        >
          <label
            class="panther-2"
          >
            <div
              class="panther-1"
            >
              <input
                aria-checked="false"
                aria-label="select item"
                class="panther-0"
                type="checkbox"
              />
            </div>
          </label>
        </div>
        <div
          class="panther-24"
        >
          <header
            class="panther-14"
          >
            <h4
              class="panther-5"
            >
              <a
                class="panther-4"
                href="/log-analysis/data-models/97c37f76-8bd8-4def-b4ab-7bfe83d62081/"
              >
                collaboration
              </a>
            </h4>
            <div
              class="panther-7"
            >
              <span
                class="panther-6"
              >
                ID
              </span>
              97c37f76-8bd8-4def-b4ab-7bfe83d62081
            </div>
            <span
              class="panther-8"
            >
              â€¢
            </span>
            <div
              class="panther-7"
            >
              <span
                class="panther-6"
              >
                Updated
              </span>
              2020-08-20 04:58 GMT
            </div>
            <div
              class="panther-13"
            >
              <button
                aria-controls="menu--4"
                aria-haspopup="true"
                aria-label="Toggle Options"
                aria-pressed="false"
                class="panther-12"
                data-reach-menu-button=""
                id="menu-button--menu--4"
                type="button"
              >
                <svg
                  class="panther-11"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 10C4.9 10 4 10.9 4 12C4 13.1 4.9 14 6 14C7.1 14 8 13.1 8 12C8 10.9 7.1 10 6 10ZM18 10C16.9 10 16 10.9 16 12C16 13.1 16.9 14 18 14C19.1 14 20 13.1 20 12C20 10.9 19.1 10 18 10ZM12 10C10.9 10 10 10.9 10 12C10 13.1 10.9 14 12 14C13.1 14 14 13.1 14 12C14 10.9 13.1 10 12 10Z"
                  />
                </svg>
              </button>
            </div>
          </header>
          <div
            class="panther-23"
          >
            <dl
              class="panther-20"
            >
              <dt
                aria-labelledby="log-typeobject-object"
                class="panther-15"
              >
                Log Type
              </dt>
              <dd
                aria-labelledby="log-typeobject-object"
                class="panther-19"
              >
                <div
                  class="panther-18"
                >
                  <span
                    class="panther-16"
                  />
                  <span
                    class="panther-17"
                  >
                    enterprise
                  </span>
                </div>
              </dd>
            </dl>
            <div
              class="panther-22"
            >
              <div
                aria-atomic="true"
                class="panther-21"
                role="status"
              >
                DISABLED
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <footer
    id="footer"
  />
</div>
`;
